### PR TITLE
Add support for frontmatter overrides

### PR DESCRIPTION
**Pull request by mheap**
### Summary
Allow setting frontmatter values for specific versions

### Reason
Needed to mark some versions as `alpha` and later versions as non-alpha.

### Testing
/kubernetes-ingress-controller/2.5.x/guides/using-gateway-api/
/kubernetes-ingress-controller/2.6.x/guides/using-gateway-api/

### CODE REVIEW

File: docs/single-sourced-versions.md [hunk 1]
```diff
@@ -183,4 +185,37 @@ If using `if_version` in a sentence, specify `inline:true` like so:
 
 ```
 Hello {% if_version eq:1.0.0 inline:true %}everyone in the {% endif_version %} world
+```
+
+### Frontmatter
```
Comment (lena-larionova) [inline docs/single-sourced-versions.md:190]:
```suggestion
### Front matter
```

File: docs/single-sourced-versions.md [hunk 2]
```diff
@@ -183,4 +185,37 @@ If using `if_version` in a sentence, specify `inline:true` like so:
...
+
+You may want to set values in the frontmatter conditionally. You can do this using `overrides`:
```
Comment (lena-larionova) [inline docs/single-sourced-versions.md:192]:
```suggestion
You may want to set values in the front matter conditionally. You can do this using `overrides`:
```

File: src/kubernetes-ingress-controller/guides/using-gateway-api.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
 ---
 title: Using Gateway API
-alpha: true
+
+overrides:
+  alpha:
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/guides/using-gateway-api.md:5]: Is the feature now in beta, or did it go straight from tech preview/alpha to GA?

**Review by lena-larionova — APPROVED** [review]
LGTM, it definitely works. Couple of minor spelling edits + a question about stability of the Gateway API, which doesn't actually need to be answered in this PR.

**Review by fabianrbz — APPROVED** [review]
LGTM

File: src/kubernetes-ingress-controller/guides/using-gateway-api.md (hunk 3 above)
Comment (mheap) [reply src/kubernetes-ingress-controller/guides/using-gateway-api.md]: I removed the changes from Gateway API and will let the KIC team update the file.

**Pull request merged**